### ISSUE TITLE
500 internal server error when accessing /sub/pages

### ISSUE THREAD

**Issue by Lesik** [picocms/Pico#226]
Hey, I would like to config Pico to use some other folder for content. I found in https://github.com/picocms/Pico/pull/211 that `content_dir` contains the content directory. Though setting `$config['content_dir'] = 'content';` does not seem to do the trick (I only see the menu, no content).

Also, this is undocumented, at least I found nothing on the docs. It's important in my opinion, would add an official way of changing the directory name, as different people use different folder structures to organize their servers.

**Comment by theshka** [comment]
I'm unable to reproduce this. 
-I did a fresh clone of the repo
-Installed and ran Composer
-Un-commented the `$config['content_dir'] = 'content-sample';` line and changed it to `$config['content_dir'] = 'content';`
-Saved the changes & renamed the `config.php.template` file to `config.php`
-Renamed the `content-sample` dir to `content`

I have a feeling you didn't rename the `config.php.template` to `config.php` for the changes to take effect. If this doesn't resolve your issue; please provide more details, and I will assist you further.

**Comment by Lesik** [comment]
Oh man, I'm sorry. I didn't notice `content_dir` was already set in the `config.php`, so I set it twice.
Yes, I did rename `config.php.template`, I'm not quite *that* stupid ;-)
Sorry, my bad, no issue here.

I face a different issue though, I can't access any files in the subdirectories, for example /sub.
`If you want another page within the sub folder, simply create a text file with the corresponding name (e.g. content-sample/sub/page.md) and will be able to access it from the URL http://yoursite.com/sub/page.`
Doesn't work for me, I just get a blank page. Tried setting chmod to 777, didn't help.

**Comment by theshka** [comment]
Sorry, @Lesik ,

Again, I'm unable to reproduce that issue. Have a look at my [test](http://dev.heshka.com/test/Pico/sub/page). A blank page usually means you have error reporting suppressed in your `php.ini` or `.htaccess` 

Check the servers' error log and respond with message it is giving.

**Comment by Lesik** [comment]
And again this issue was my fault. I'm starting to feel embarrassed.
I though I copied the .htaccess file when transferring my local folder to the server, but because "hidden files" were turned off, Nautilus (my file manager) didn't copy that file over.

`6. That's it. Tweak the .htaccess file if required.`
With the default .htaccess (from this repo) I get a 500 server error accessing /sub. Could you share your .htaccess for comparison?

**Comment by theshka** [comment]
This is the one my test case is using, just the default `.htaccess`

```
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d 
RewriteRule . index.php [L]
</IfModule>

# Prevent file browsing
Options -Indexes
```

**Comment by Lesik** [comment]
I have the same one. I get this error when accessing /sub:
`Internal Server Error

The server encountered an internal error or misconfiguration and was unable to complete your request.

Please contact the server administrator, and inform them of the time the error occurred, and anything you might have done that may have caused the error.

More information about this error may be available in the server error log.

Additionally, a 500 Internal Server Error error was encountered while trying to use an ErrorDocument to handle the request.`

I don't have SSH access on my server, just SFTP (German hoster 1und1.de), so can't check logs. Any idea what could be wrong anyways..? :(

**Comment by theshka** [comment]
I'm not familiar with the host, and It's hard for me to say.  I know that 500 errors are typical of a `.htaccess` misconfiguration. :stuck_out_tongue_winking_eye: 

- Double-check the file permissions. 
- Is your Pico installation in a subfolder? Have you tried [RewriteBase](http://httpd.apache.org/docs/current/mod/mod_rewrite.html#rewritebase)? 
- I'd also echo a `phpinfo();` and make sure the server configuration is proper. 

If nothing else, try submitting a ticket to your host. Having access to the log file could give you crucial information.

**Comment by Lesik** [comment]
- What permissions should be set? Usually CMSs want full 777, but I haven't seen anything in the Pico docs
- It's not in a subfolder, but on a subdomain. Would I need `RewriteBase`?
- `phpinfo();` results in a giant output, what should I look for?

I know how difficult it is to deal with inexperienced people, thanks for having patience with me. Webdev isn't my field, I just want to host my Java projects with a simple CMS.

**Comment by theshka** [comment]
It's no problem. I'm no master myself, and this helps me just as much as it help you. 

- Generally speaking, files: 644 & folders: 755. 
- It shouldn't, since the subdomain would point to the document root, but It really depends on your host and their configuration. Try throwing a `RewriteBase /` <- notice the trailing `/` and see if that fixes things, if not, revert. 
- Basic things: Is the server running PHP 5.3+? Is `mod_rewrite` listed under `Loaded Modules`? This also provides you with a lot of information on how the server is configured, and what mods/extensions are available to use. Shared hosting can be a bitch. 

Have you tried this in a local environment?

**Comment by theshka** [comment]
You may get more information from these issues as well: #100, #22

**Comment by Lesik** [comment]
I did a `Ctrl+F` for mod_rewrite and couldn't find it. I tried taking out the `<IfModule mod_rewrite.c>`. I don't know if I understand Apache configuration correctly, but this should disable the checking for mod_rewrite and do `RewriteEngine On` independant of if mod_rewrite is there or not. If mod_rewrite is not there, it should result in a 500 server error, right? But I don't and suddenly the /sub directory is accessible too.
